### ISSUE TITLE
[no-unnecessary-condition] with optional argument and optional chain

### ISSUE THREAD

**Issue by dangreen** [typescript-eslint/typescript-eslint#2333]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unnecessary-condition": ["error",  {
      "allowConstantLoopConditions": true
    }]
  }
}
```

```TS
function getProperties(property: string, exclude?: RegExp) {
  return exclude?.test(property)
}
```

**Expected Result**

No suggestions

**Actual Result**

> error  Unnecessary optional chain on a non-nullish value  @typescript-eslint/no-unnecessary-condition

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.2.0` |
| `@typescript-eslint/parser`        | `3.2.0` |
| `TypeScript`                       | `3.9.6` |
| `ESLint`                           | `7.4.0` |
| `node`                             | `12.14.1` |
| `npm`                              | `6.13.4` |

**Comment by dangreen** [comment]
Code from example also invalid:

https://github.com/typescript-eslint/typescript-eslint/blob/v3.2.0/packages/eslint-plugin/docs/rules/no-unnecessary-condition.md

```ts
function bar(arg?: string | null) {
  return arg?.length // Unnecessary optional chain on a non-nullish value 
}
```

**Comment by bradzacher** [comment]
could you please try this against a newer version? Latest version is 3.7.1, but you're on 3.2.0

**Comment by dangreen** [comment]
@bradzacher Hi. `strictNullChecks` helps, thank you! Will be nice have info about that in rule docs.
